back/spec/policies/initiative_policy_spec.rb

```diff
@@ -0,0 +1,300 @@
+          end
+
+          context 'for a user who is not author of the initiative' do
+            let(:user) { create(:user) }
+
+            it { is_expected.not_to permit(:show) }
+            it { is_expected.not_to permit(:by_slug) }
+            it { is_expected.not_to permit(:create) }
+            it { is_expected.not_to permit(:update)  }
+            it { is_expected.not_to permit(:destroy) }
+
+            it 'does not index the initiative' do
+              expect(scope.resolve.size).to eq 0
+            end
+          end
+
+          # Author can see initiative, but not edit, if approval feature is active.
+          context 'for a user who is author of the initiative' do
+            let(:user) { author }
+
+            it { is_expected.to permit(:show) }
+            it { is_expected.to permit(:by_slug) }
+            it { is_expected.not_to permit(:create) }
+            it { is_expected.not_to permit(:update)  }
+            it { is_expected.not_to permit(:destroy) }
+
+            it 'indexes the initiative' do
+              expect(scope.resolve.size).to eq 1
+            end
+          end
+
+          context 'for a visitor' do
+            let(:user) { nil }
+
+            it { is_expected.not_to permit(:show) }
+            it { is_expected.not_to permit(:by_slug) }
+            it { expect { policy.create? }.to raise_error(Pundit::NotAuthorizedError) }
+            it { expect { policy.update? }.to raise_error(Pundit::NotAuthorizedError) }
+            it { expect { policy.destroy? }.to raise_error(Pundit::NotAuthorizedError) }
+
+            it 'does not index the initiative' do
+              expect(scope.resolve.size).to eq 0
+            end
+          end
+        end
+      end
+    end
+
+    context 'for initiatives with NOT_APPROVAL_CODES statuses' do
+      (InitiativeStatus::NOT_APPROVAL_CODES - ['custom']).each do |code|
+        context "for an initiative with #{code} status" do
+          let!(:initiative) do
+            create(:initiative, author: author, initiative_status: create("initiative_status_#{code}".to_sym))
+          end
+
+          context 'for a user who is not author of the initiative' do
+            let(:user) { create(:user) }
+
+            it { is_expected.to permit(:show) }
+            it { is_expected.to permit(:by_slug) }
+            it { is_expected.not_to permit(:create) }
+            it { is_expected.not_to permit(:update)  }
+            it { is_expected.not_to permit(:destroy) }
+
+            it 'does not index the initiative' do
+              expect(scope.resolve.size).to eq 1
+            end
+          end
+
+          # Author can see initiative, but not edit, if approval feature is active.
+          context 'for a user who is author of the initiative' do
+            let(:user) { author }
+
+            it { is_expected.to permit(:show) }
+            it { is_expected.to permit(:by_slug) }
+            it { is_expected.not_to permit(:create) }
```

Comment:
Why can't this user create an initiative? How could initiatives be created in this case?